enterprise/views.py

```diff
@@ -771,7 +771,16 @@ def post(self, request, enterprise_uuid, course_id):
                 ),
                 query_string=urlencode({'course_mode': selected_course_mode_name})
             )
-            failure_url = reverse('enterprise_course_enrollment_page', args=[enterprise_customer.uuid, course_id])
+            if request.META['QUERY_STRING']:
+                failure_url = '{course_enrollment_url}?{query_string}'.format(
+                    course_enrollment_url=reverse(
+                        'enterprise_course_enrollment_page', args=[enterprise_customer.uuid, course_id]
+                    ),
+                    query_string=request.META['QUERY_STRING']
+                )
+            else:
+                failure_url = reverse('enterprise_course_enrollment_page', args=[enterprise_customer.uuid, course_id])
```

Comment:
Let's tighten this up a bit:
```
failure_url = reverse('enterprise_course_enrollment_page', args=[enterprise_customer.uuid, course_id])
if request.META['QUERY_STRING']:
    failure_url = '{course_enrollment_url}?{query_string}'.format(
        course_enrollment_url=failure_url,
        query_string=request.META['QUERY_STRING']
    )
```

Reply:
Also, I think we could use a comment explaining what is going on here.

Reply:
Done.